common/src/main/java/io/honeycomb/opentelemetry/sdk/trace/samplers/DeterministicTraceSampler.java

```diff
@@ -78,10 +98,18 @@ public SamplingResult shouldSample(
         Attributes attributes,
         List<LinkData> parentLinks) {
 
-        SamplingResult result = baseSampler.shouldSample(parentContext, traceId, name, spanKind, attributes, parentLinks);
-        return SamplingResult.create(
-            result.getDecision(),
-            Attributes.of(SAMPLE_RATE_ATTRIBUTE_KEY, (long) sampleRate)
-        );
+        SamplingResult result =
```

Comment:
Looks like we're using the decoratedSampler then the baseSampler but the the comments indicate we expect the basSampler to execute first, then the decorated sampler.

This need to be consistent to ensure we get the expected behaviour.

Reply:
Yes! Good pick up.

The implementation ended up this way because I figured it made sense to always include the `decoratedSampler`'s attributes, regardless of what the decision was. `baseSampler`, on the other hand, we know delivers no attributes in its result, so we lose nothing by skipping it.

I'll update the comment to match the implementation.